File: WitchcraftApp/src/test/java/com/tarbadev/witchcraft/domain/usecase/SetFavoriteRecipeUseCaseTest.java
package com.tarbadev.witchcraft.domain.usecase;

import com.tarbadev.witchcraft.domain.repository.RecipeRepository;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.junit.MockitoJUnitRunner;

import static org.mockito.Mockito.verify;

@RunWith(MockitoJUnitRunner.class)
public class SetFavoriteRecipeUseCaseTest {
  @Mock
  private RecipeRepository recipeRepository;

  private SetFavoriteRecipeUseCase subject;

  @Before
  public void setUp() {
    subject = new SetFavoriteRecipeUseCase(recipeRepository);
  }

  @Test
  public void execute() {
    subject.execute(123, true);

    verify(recipeRepository).setFavorite(123, true);
  }
}

File: WitchcraftApp/src/test/java/com/tarbadev/witchcraft/recipes/domain/usecase/SetFavoriteRecipeUseCaseTest.kt
package com.tarbadev.witchcraft.recipes.domain.usecase

import com.nhaarman.mockitokotlin2.mock
import com.tarbadev.witchcraft.recipes.domain.repository.RecipeRepository
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.mockito.Mockito.verify

class SetFavoriteRecipeUseCaseTest {
    private val recipeRepository: RecipeRepository = mock()
    private lateinit var subject: SetFavoriteRecipeUseCase

    @BeforeEach
    fun setUp() {
        subject = SetFavoriteRecipeUseCase(recipeRepository)
    }

    @Test
    fun execute() {
        subject.execute(123, true)

        verify<RecipeRepository>(recipeRepository).setFavorite(123, true)
    }
}